server/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTests.java

```diff
@@ -93,16 +92,18 @@ private static void assertParseSearchSource(SearchSourceBuilder testBuilder, XCo
     }
 
     public void testSerialization() throws IOException {
-        SearchSourceBuilder testBuilder = createSearchSourceBuilder();
-        try (BytesStreamOutput output = new BytesStreamOutput()) {
-            testBuilder.writeTo(output);
-            try (StreamInput in = new NamedWriteableAwareStreamInput(output.bytes().streamInput(), namedWriteableRegistry)) {
-                SearchSourceBuilder deserializedBuilder = new SearchSourceBuilder(in);
-                assertEquals(deserializedBuilder, testBuilder);
-                assertEquals(deserializedBuilder.hashCode(), testBuilder.hashCode());
-                assertNotSame(deserializedBuilder, testBuilder);
-            }
-        }
+        SearchSourceBuilder original = createSearchSourceBuilder();
+        SearchSourceBuilder copy = copyBuilder(original);
+        assertEquals(copy, original);
+        assertEquals(copy.hashCode(), original.hashCode());
+        assertNotSame(copy, original);
+    }
+
+    public void testSerializingWithRuntimeFieldsBeforeSupportedThrows() {
+        SearchSourceBuilder original = new SearchSourceBuilder().runtimeMappings(randomRuntimeMappings());
+        Version v = Version.CURRENT.minimumCompatibilityVersion();
```

Comment:
Maybe call this out as Version.V_8_0_0 explicitly so it doesn't start failing once master is on v9?

Reply:
When we go to v9.0.0 we'll want to remove the test anyway. Frankly, I think we'll blast the test as soon as I'm done backporting because we'll be able to send to 7.11.0. This `v` will be a bit more complex in the backport.